engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java

```diff
@@ -321,11 +334,73 @@ public void shouldFailJobWithRetriesLeft() {
     jobState.fail(key, jobRecord);
 
     // then
-    Assertions.assertThat(jobState.exists(key)).isTrue();
+    assertThat(jobState.exists(key)).isTrue();
     assertJobState(key, State.ACTIVATABLE);
     assertJobRecordIsEqualTo(jobState.getJob(key), jobRecord);
     assertListedAsActivatable(key, jobRecord.getTypeBuffer());
     refuteListedAsTimedOut(key, jobRecord.getDeadline() + 1);
+    refuteListedAsBackOff(key, jobRecord.getRecurringTime() + 1 + 1);
+  }
+
+  @Test
+  public void shouldFailJobWithRetriesAndBackOff() {
+    // given
+    final long key = 1L;
+    final var retryBackoff = 100;
+    final JobRecord jobRecord = newJobRecord().setRetries(1).setRetryBackoff(retryBackoff);
+
+    // when
+    jobState.create(key, jobRecord);
+    jobState.activate(key, jobRecord);
+    jobState.fail(key, jobRecord);
+
+    // then
+    assertThat(jobState.exists(key)).isTrue();
+    assertJobState(key, State.FAILED);
+    assertJobRecordIsEqualTo(jobState.getJob(key), jobRecord);
+    refuteListedAsActivatable(key, jobRecord.getTypeBuffer());
+    refuteListedAsTimedOut(key, jobRecord.getDeadline() + 1);
+    assertListedAsBackOff(key, jobRecord.getRecurringTime() + 1 + retryBackoff);
+  }
+
+  @Test
+  public void shouldRetryProperJobWithRetryBackoff() {
+    // given
+    final long firstKey = 1L;
+    final long secondKey = 2L;
+    final JobRecord firstJobRecord = newJobRecord().setRetries(1).setRetryBackoff(0);
+    final JobRecord secondJobRecord = newJobRecord().setRetries(1).setRetryBackoff(0);
+
+    // when
+    jobState.create(firstKey, firstJobRecord);
+    jobState.activate(firstKey, firstJobRecord);
+    jobState.create(secondKey, secondJobRecord);
+    jobState.activate(secondKey, secondJobRecord);
+    jobState.fail(firstKey, firstJobRecord);
+    jobState.fail(secondKey, secondJobRecord);
+
+    // then
+    assertThat(jobState.exists(firstKey)).isTrue();
+    assertThat(jobState.exists(secondKey)).isTrue();
+    assertJobState(firstKey, State.ACTIVATABLE);
+    assertJobState(secondKey, State.ACTIVATABLE);
+  }
+
+  @Test
+  public void shouldRetryJobAfterRecurAfterBackoff() {
```

Comment:
:x: We should add another test case for `recurAfterBackoff()`.

Currently, we don't have a test case to verify that `recurAfterBackoff()` makes the job activatable again and removes it from the back-off column.